inference-engine/src/mkldnn_plugin/nodes/gather.cpp

```diff
@@ -22,18 +22,41 @@ namespace InferenceEngine {
 namespace Extensions {
 namespace Cpu {
 
+using MKLDNNPlugin::TensorDescCreatorTypes;
+
 class GatherImpl: public ExtLayerBase {
 public:
+    static bool isSupportedParams(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
```

Comment:
Are we sure we need this method to be `noexcept`? As far as I can see, this method is used in the GatherImpl constructor that throws exceptions. Eliminating this restriction would allow to use a simpler error message passing (through exceptions).